core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitor.java

```diff
@@ -259,69 +251,121 @@ private String retrievePeerTableName() {
     return "system.peers";
   }
 
-  private NodeInfo asNodeInfo(AdminRow row) {
-    return nodeInfoBuilder(row, getBroadcastRpcAddress(row)).build();
-  }
-
-  private Optional<NodeInfo> firstRowAsNodeInfo(AdminResult result) {
+  private Optional<NodeInfo> firstPeerRowAsNodeInfo(AdminResult result) {
     Iterator<AdminRow> iterator = result.iterator();
     if (iterator.hasNext()) {
-      return Optional.of(asNodeInfo(iterator.next()));
+      return buildNodeInfo(iterator.next());
     } else {
       return Optional.empty();
     }
   }
 
   /**
-   * @param broadcastRpcAddress this is a parameter only because we already have it when we come
-   *     from {@link #findInPeers(AdminResult, InetSocketAddress)}. Callers that don't already have
-   *     it can use {@link #getBroadcastRpcAddress}.
+   * Creates a {@link NodeInfo} instance from the given row; returns empty if the row does not
+   * contain enough information, in which case the node will be discarded and won't show up in the
+   * nodes list.
    */
-  protected DefaultNodeInfo.Builder nodeInfoBuilder(
-      AdminRow row, InetSocketAddress broadcastRpcAddress) {
+  @NonNull
+  protected Optional<NodeInfo> buildNodeInfo(@NonNull AdminRow row) {
```

Comment:
I had to remove the second parameter because for a peer row, we need first to validate the row and only then obtain the broadcast RPC address.
On the flip side, the method signature becomes more straightforward, which is probably a good think given that this method is meant to be overridden (see DSE driver).